analyses/focal-cn-file-preparation/01-prepare-cn-file.R

```diff
@@ -263,20 +285,73 @@ cnv_no_xy <- cnv_df %>%
 
 # Merge and annotated no X&Y
 autosome_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_no_xy,
-                                                   txdb_exons = tx_exons) %>%
+                                                   txdb_exons = exons_gr) %>%
   # mark possible amplifications in autosomes
   dplyr::mutate(status = dplyr::case_when(
     copy_number > (2 * ploidy) ~ "amplification",
     TRUE ~ status
   ))
 
+#### Wrangle GISTIC data ---------------------------------------------
+
+# Transpose the GISTIC data.frame
+autosome_transposed_gistic_df <- gistic_df %>%
+  tibble::column_to_rownames("Chromosome Arm") %>%
+  t() %>%
+  as.data.frame() %>%
+  dplyr::select(-c("Xp", "Xq", "Yp", "Yq")) %>%
+  tibble::rownames_to_column("biospecimen_id")
+
+# Gather the chromosome arm data into one column and the GISTIC call in another
+autosome_transposed_gistic_df <- autosome_transposed_gistic_df %>%
+  dplyr::mutate(biospecimen_id_copy = biospecimen_id) %>%
+  tidyr::gather(key = "arm", value = "gistic_call", -biospecimen_id) %>%
+  dplyr::distinct()
+
+# Filter the GISTIC data to only the overlapping samples
+autosome_filtered_gistic_df <- autosome_transposed_gistic_df %>%
+  dplyr::filter(biospecimen_id %in% autosome_annotated_cn$biospecimen_id, 
+                !(arm == "biospecimen_id_copy")) %>%
+  dplyr::mutate(broad_status = dplyr::case_when(gistic_call == -1 ~ "loss",
+                                                gistic_call == 0 ~ "neutral",
+                                                gistic_call == 1 ~ "gain",
+                                                TRUE ~ "amplification")) %>%
+  dplyr::select(-gistic_call)
+
+# Add GISTIC data to final data.frame
+autosome_annotated_cn <- autosome_annotated_cn %>%
+  dplyr::inner_join(autosome_filtered_gistic_df, by = "biospecimen_id")
+
 # Output file name
 autosome_output_file <- paste0(opt$filename_lead, "_autosomes.tsv")
 
 # Save final data.frame to a tsv file
 readr::write_tsv(autosome_annotated_cn,
                  file.path(results_dir, autosome_output_file))
 
+#### Compare `org.Hs.eg.db` package and UCSC cytoband calls -------------------
+
+# Data.frame comparing cytoband calls
```

Comment:
I like having this on record somewhere, but I'm not sure if we wanna keep it here.

Reply:
I was thinking the same, it would be recorded in the commit history, but I am not sure if I should break it out into a separate file as well.

Reply:
Meh. Maybe put a lengthy comment on where we read in the UCSC cytobands file to explain the choice. Between that and the README, should be fine??

Reply:
Sounds good to me.